### PR TITLE
add ability to go offline

### PR DESCRIPTION
**Pull request by djih**
Similar to: https://github.com/amplitude/Amplitude-Android/blob/master/src/com/amplitude/api/AmplitudeClient.java#L223-L232

### CODE REVIEW

File: AmplitudeTests/AmplitudeTests.m [hunk 1]
```diff
@@ -368,4 +368,30 @@ -(void)testAutoIncrementSequenceNumber {
     }
 }
 
+-(void)testSetOffline {
```
Comment (curtisliu) [inline AmplitudeTests/AmplitudeTests.m]: can you add a test that events still get truncated when offline?

Comment (djih) [inline AmplitudeTests/AmplitudeTests.m]: added test and found a bug!

**Comment by djih** [comment]
Fixed bug where events weren't being dropped if eventMaxCount < eventRemoveBatchSize (since getting nthEventId for eventRemoveBatchSize would return -1). Added tests for truncating events and truncating events while offline. PTAL @curtisliu

File: Amplitude/Amplitude.m [hunk 2]
```diff
@@ -533,6 +531,20 @@ - (void)logEvent:(NSString*) eventType withEventProperties:(NSDictionary*) event
     }];
 }
 
+- (void)truncateEventQueues
+{
+    AMPDatabaseHelper *dbHelper = [AMPDatabaseHelper getDatabaseHelper];
+    int numEventsToRemove = self.eventMaxCount > kAMPEventRemoveBatchSize ? kAMPEventRemoveBatchSize : self.eventMaxCount/2;
```
Comment (curtisliu) [inline Amplitude/Amplitude.m]: how did you choose self.eventMaxCount / 2?

Comment (curtisliu) [inline Amplitude/Amplitude.m]: I think this should be more like min(max(1, self.eventMaxCount / 10), kAMPEventRemoveBatchSize)

Comment (djih) [inline Amplitude/Amplitude.m]: Changed

**Comment by curtisliu** [comment]
LGTM

**Pull request merged**